### ISSUE TITLE
Add unit test to verify selectors/mdqe return expected values

### ISSUE THREAD

**Issue by gothub** [NCEAS/mdqengine#52]
Create a suite that contains checks to validate that the selectors of a check
deliver expected data types and values to the check and are represented correctly
in the check's language. Some possible types/xpaths to check:

- logical, i.e. boolean()
- numeric types, i.e. count()
- character types, i.e. string()
- node types, i.e. text(), name()
- selecting element attributes

The current set of suites are using some of these, but it would be very useful to have
a suite that tests the selected values explicitly, and possibly in all supported check languages.

**Comment by leinfelder** [comment]
This would be similar to PythonDispatcherTest.testTypes() but would include XMLDialect for selecting xpaths from a known document?

**Comment by gothub** [comment]
Yes, exactly! Except there would be no Java code involved, just a check suite, first written in
R (our favorite language). The checks would simply test the data types and values
delivered to the checks. We have been doing this in the checks already, but it would be
really useful to gather and extend these checks into one suite.

**Comment by leinfelder** [comment]
Junit test with no Java code...wut?

**Comment by gothub** [comment]
Thanks @leinfelder, I changed the issue title which was in error. This is not 
intended to be a unit test, more of an informal integration test.
